app/src/main/java/com/github/studydistractor/sdp/account/FirebaseCreateAccount.kt

```diff
@@ -12,60 +12,27 @@ class FirebaseCreateAccount @Inject constructor(): CreateAccountInterface{
 
     private var auth = FirebaseAuth.getInstance()
 
-    private var db: FirebaseDatabase = FirebaseDatabase.getInstance()
-
-    private var PATH = "Users"
-
-
-
-    /**
-     * Check if the given phone number is not empty and its length is exactly 10 characters
-     * @param phoneNumber The phone number to check
-     * @return true if the phone number is correctly formatted
-     */
-    @Inject
-    override fun checkPhoneFormat(phoneNumber: String): Boolean {
-        return !(phoneNumber.isEmpty() || phoneNumber.length != 10)
-    }
-
     /**
-     * Check if the given name is not empty and it does not exceed or is equal to 255 characters
-     * @param name The name to check
-     * @return true if the name is correctly formatted
-     */
-    @Inject
-    override fun checkNameFormat(name: String): Boolean {
-        return !(name.isEmpty() || name.length >= 255)
-    }
-
-
-    /**
-    * Method to create a user account in Firebase database.
-    * @param year the user's year of birth
-    * @param month the user's month of birth
-    * @param day the user's day of birth
-    * @param firstname the user's first name
-    * @param lastname the user's last name
-    * @param phone the user's phone number
+     * Method to create a user account in Firebase database.
+     * @param year the user's year of birth
+     * @param month the user's month of birth
+     * @param day the user's day of birth
+     * @param firstname the user's first name
+     * @param lastname the user's last name
+     * @param phone the user's phone number
+     * @
     */
     @Inject
-    override fun postData(year: Int, month: Int, day: Int, firstname: String, lastname: String, phone: String){
+    override fun createUser(
+        birthday: HashMap<String,Int>,
```

Comment:
Maybe create a object for the birthday date. Like object Date {day : Int, mouth : Int, year : Int} not a HashMap that can be of any struct.

Reply:
I've tried but Firebase does not succed to upload it on the database